sources/command/boot.py

```diff
@@ -3,38 +3,48 @@
 
 from globals import ExitCode, get_devices
 from systemd import Systemd
+from openrc import Openrc
+
 
 def boot(boot_status: str) -> NoReturn:
-    """Enable or disable the systemd service which 
-        activates the ir emitter for all configured device
+    """Enable or disable the systemd service which
+    activates the ir emitter for all configured device
 
-        args:
-            boot_status: "enable" or "disable" or "status"
+    args:
+        boot_status: "enable" or "disable" or "status"
 
-        Raises:
-            Exception: boot status arg can only be equal to 
-            enable, disable or status
+    Raises:
+        Exception: boot status arg can only be equal to
+        enable, disable or status
     """
-    assert(boot_status in ["enable", "disable", "status"])
+    assert boot_status in ["enable", "disable", "status"]
+    systemd_mode = (
+        subprocess.run(["which", "systemctl"], capture_output=True).returncode == 0
+    )
 
     if boot_status == "enable":
         devices = get_devices()
         if not len(devices):
             logging.critical("No driver have been configured.")
             exit(ExitCode.FAILURE)
 
-        systemd = Systemd(devices)
-        if systemd.enable():
+        if systemd_mode and Systemd(devices).enabled():
```

Comment:
There is a type here, this is `enable()`.

Reply:
Also, for those using openrc, this will display error messages logged by `Systemd.enable`
that can lead to the belief that something is not working, while everything is under control.